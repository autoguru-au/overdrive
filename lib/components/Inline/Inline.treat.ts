import { style } from 'treat';

import { makeResponsiveStyle } from '../../utils';

const flexShared = style(
	{
		display: 'flex',
	},
	'flexShared',
);

export const root = [
	flexShared,
	style(
		{
			justifyContent: 'flex-start',
			flexWrap: 'wrap',
		},
		'root',
	),
];

const alignRules = {
	top: 'flex-start',
	center: 'center',
	bottom: 'flex-end',
};

export const align = makeResponsiveStyle(() => alignRules, 'alignItems');

export const child = [flexShared];
