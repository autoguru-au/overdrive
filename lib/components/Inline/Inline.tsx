import clsx from 'clsx';
import * as React from 'react';
import { Children, FunctionComponent } from 'react';
import { useStyles } from 'react-treat';
import { Theme } from 'treat/theme';

import {
	useNegativeMarginLeft,
	useNegativeMarginTop,
} from '../../hooks/useNegativeMargin/useNegativeMargin';
import { resolveResponsiveStyle, ResponsiveProp } from '../../utils';
import { Box } from '../Box';
import * as styleRefs from './Inline.treat';

interface Props {
	space?: ResponsiveProp<keyof Theme['space']>;
	alignY?: ResponsiveProp<keyof typeof styleRefs.align>;
}

export const Inline: FunctionComponent<Props> = ({
	children,
	space = '2',
	alignY,
}) => {
	const negativeMarginLeft = useNegativeMarginLeft(space);
	const negativeMarginTop = useNegativeMarginTop(space);

	const styles = useStyles(styleRefs);

	return (
		<Box className={negativeMarginTop}>
			<Box
				className={clsx(
					styles.root,
					negativeMarginLeft,
					resolveResponsiveStyle(alignY, styles.align),
				)}>
				{Children.map(children, (child) => (
					<Box
						display="inline-block"
						paddingLeft={space}
						paddingTop={space}>
						{child}
					</Box>
				))}
			</Box>
		</Box>
	);
};
