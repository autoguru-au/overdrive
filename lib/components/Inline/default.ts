export { Inline as default } from './Inline';
