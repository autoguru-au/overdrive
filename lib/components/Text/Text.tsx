import cx from 'clsx';
import React, { FunctionComponent } from 'react';
import styles from './style.scss';

export interface IProps {
	variant?: EVariant | null;
	className?: string;
	muted?: boolean;
}

export enum EVariant {
	Mobile = 'mobile',
	Desktop = 'desktop',
}

export const Text: FunctionComponent<IProps> = ({
	variant = null,
	className = '',
	muted = false,
	children,
}) => (
	<p
		className={cx([styles.root, className], {
			[styles.muted]: muted,
			[styles.variantMobile]: !!variant && variant === EVariant.Mobile,
			[styles.variantDesktop]: !!variant && variant === EVariant.Desktop,
		})}
		children={children}
	/>
);
