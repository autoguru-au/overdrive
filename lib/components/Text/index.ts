export { Text } from './Text';
export { useTextStyles } from './useTextStyles';

export type { TextStyleProps } from './useTextStyles';
