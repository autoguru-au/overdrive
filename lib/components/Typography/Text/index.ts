export { Text } from './Text';
export { useTextStyles } from './useTextStyles';
