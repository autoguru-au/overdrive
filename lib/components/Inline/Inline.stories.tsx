import { Meta, StoryObj } from '@storybook/react';
import React from 'react';

// import { scaleOptions } from '../Box/argTypes';
import { Button } from '../Button';
import { Stack } from '../Stack';
import { Text } from '../Text';

import { Inline } from '.';

const meta: Meta<typeof Inline> = {
	title: 'Layout/Inline',
	component: Inline,
	// argTypes: {
	// 	alignY: {
	// 		options: ['flexStart', 'center', 'flexEnd'],
	// 		control: {
	// 			type: 'select',
	// 		},
	// 	},
	// 	alignX: {
	// 		options: ['flexStart', 'center', 'flexEnd', 'spaceBetween'],
	// 		control: {
	// 			type: 'select',
	// 		},
	// 	},
	// 	space: {
	// 		options: scaleOptions,
	// 		control: {
	// 			type: 'select',
	// 		},
	// 	},
	// 	dividers: {
	// 		control: {
	// 			type: 'boolean',
	// 		},
	// 	},
	// },
};

export default meta;

type Story = StoryObj<typeof Inline>;

export const Standard: Story = {
	args: {},
	render: () => (
		<Inline>
			<Text>Mazda</Text>
			<Text>CX3</Text>
			<Text>Petrol</Text>
			<Text>2020</Text>
		</Inline>
	),
};

export const Dividers: Story = {
	args: {
		dividers: true,
	},
	render: (args) => (
		<Inline {...args}>
			<Text>Mazda</Text>
			<Text>CX3</Text>
			<Text>Petrol</Text>
			<Text>2020</Text>
		</Inline>
	),
};

export const CustomDividers: Story = {
	args: {
		dividers: (
			<div
				style={{
					backgroundColor: 'red',
					width: '6px',
					height: '6px',
				}}
			/>
		),
	},
	render: (args) => (
		<Inline {...args}>
			<Text>Mazda</Text>
			<Text>CX3</Text>
			<Text>Petrol</Text>
			<Text>2020</Text>
		</Inline>
	),
};

export const DifferentSizeItems: Story = {
	args: {
		dividers: true,
	},
	render: (args) => (
		<Stack>
			<Inline {...args}>
				<Text>Mazda</Text>
				<Text>CX3</Text>
				<Text>Petrol</Text>
				<Text>2020</Text>
				<Button>Buy</Button>
			</Inline>
			<Inline {...args}>
				<Text>Mazda</Text>
				<Text>CX3</Text>
				<Text>Petrol</Text>
				<Text>2020</Text>
				<Button>Buy</Button>
			</Inline>
			<Inline {...args}>
				<Text>Mazda</Text>
				<Text>CX3</Text>
				<Text>Petrol</Text>
				<Text>2020</Text>
				<Button>Buy</Button>
			</Inline>
		</Stack>
	),
};

export const WithFullWidth: Story = {
	args: {
		width: 'full',
		alignX: 'space-between',
	},
	render: (args) => (
		<Stack>
			<Inline {...args}>
				<Text>Mazda</Text>
				<Text>CX3</Text>
				<Text>Petrol</Text>
				<Text>2020</Text>
				<Button>Buy</Button>
			</Inline>
			<Inline {...args}>
				<Text>Mazda</Text>
				<Text>CX3</Text>
				<Text>Petrol</Text>
				<Text>2020</Text>
				<Button>Buy</Button>
			</Inline>
			<Inline {...args}>
				<Text>Mazda</Text>
				<Text>CX3</Text>
				<Text>Petrol</Text>
				<Text>2020</Text>
				<Button>Buy</Button>
			</Inline>
		</Stack>
	),
};
