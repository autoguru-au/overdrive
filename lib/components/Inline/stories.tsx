import * as React from 'react';

import { Button } from '../Button';
import { Stack } from '../Stack';
import { Text } from '../Text';
import { Inline } from '.';

export default {
	title: 'Foundation|Layout/Inline',
	component: Inline,
};

export const Standard = () => (
	<Inline>
		<Text>Mazda</Text>
		<Text>CX3</Text>
		<Text>Petrol</Text>
		<Text>2020</Text>
	</Inline>
);

export const Dividers = () => (
	<Inline dividers>
		<Text>Mazda</Text>
		<Text>CX3</Text>
		<Text>Petrol</Text>
		<Text>2020</Text>
	</Inline>
);

export const CustomDividers = () => (
	<Inline
		dividers={
			<div
				style={{
					backgroundColor: 'red',
					width: '6px',
					height: '6px',
				}}
			/>
		}>
		<Text>Mazda</Text>
		<Text>CX3</Text>
		<Text>Petrol</Text>
		<Text>2020</Text>
	</Inline>
);

export const DifferentSizeItems = () => (
	<Stack>
		<Inline dividers>
			<Text>Mazda</Text>
			<Text>CX3</Text>
			<Text>Petrol</Text>
			<Text>2020</Text>
			<Button>Buy</Button>
		</Inline>
		<Inline dividers alignY="flexEnd">
			<Text>Mazda</Text>
			<Text>CX3</Text>
			<Text>Petrol</Text>
			<Text>2020</Text>
			<Button>Buy</Button>
		</Inline>
		<Inline alignY="flexStart">
			<Text>Mazda</Text>
			<Text>CX3</Text>
			<Text>Petrol</Text>
			<Text>2020</Text>
			<Button>Buy</Button>
		</Inline>
	</Stack>
);
