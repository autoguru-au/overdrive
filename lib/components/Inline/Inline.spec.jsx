import { render } from '@testing-library/react';
import * as React from 'react';

import { Text } from '../Text';

import { Inline } from './Inline';

const TestComponent = () => <div>divider component from test</div>;

describe('<Inline />', () => {
	it('should not throw', () => {
		expect(() => render(<Inline />)).not.toThrow();
	});

	it.each([
		['default', true],
		['no', false],
		['custom element', <div>divider</div>],
	])('should match snapshot for %s dividers', (label, value) => {
		const { container } = render(
			<Inline dividers={value}>
				<Text>A</Text>
				<Text>B</Text>
				<Text>C</Text>
			</Inline>,
		);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should not be able to render components', () => {
		const component = (
			<Inline dividers={<TestComponent />}>
				<Text>A</Text>
				<Text>B</Text>
			</Inline>
		);

		expect(() => render(component)).not.toThrow();

		const { container } = render(component);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should not render undefined children', () => {
		const { container } = render(
			<Inline>
				{null}
				<Text>A</Text>
				<Text>B</Text>
				<Text>C</Text>
				{false}
			</Inline>,
		);

		expect(container.firstChild.children).toHaveLength(3);
	});
});
