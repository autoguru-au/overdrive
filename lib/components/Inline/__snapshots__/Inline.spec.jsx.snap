// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inline /> should match the snapshot 1`] = `
<div
  class="base 2_mobile_base"
>
  <div
    class="base root 2_mobile_base"
  />
</div>
`;
